conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+    }
+
+    private static final class ByteArrayRequestBody implements RequestBody {
+
+        private final byte[] buffer;
+        private final String contentType;
+
+        ByteArrayRequestBody(byte[] buffer, String contentType) {
+            this.buffer = buffer;
+            this.contentType = contentType;
+        }
+
+        @Override
+        public void writeTo(OutputStream output) throws IOException {
+            output.write(buffer);
+        }
+
+        @Override
+        public String contentType() {
+            return contentType;
+        }
+
+        @Override
+        public boolean repeatable() {
+            return true;
+        }
+
+        @Override
+        public void close() {
+            // nothing to do
+        }
+    }
+
+    private static final class DialogueResponseBody implements feign.Response.Body {
+
+        private final Response response;
+
+        DialogueResponseBody(Response response) {
+            this.response = response;
+        }
+
+        @Override
+        public Integer length() {
+            return response.getFirstHeader(HttpHeaders.CONTENT_LENGTH)
+                    .map(Ints::tryParse)
+                    .orElse(null);
+        }
+
+        @Override
+        public boolean isRepeatable() {
+            return false;
+        }
+
+        @Override
+        public InputStream asInputStream() {
+            return response.body();
+        }
+
+        @Override
+        public Reader asReader() {
+            return new InputStreamReader(asInputStream(), StandardCharsets.UTF_8);
+        }
+
+        @Override
+        public void close() {
+            response.close();
+        }
+    }
+
+    enum IdentityDeserializer implements Deserializer<Response> {
+        INSTANCE;
+
+        @Override
+        public Response deserialize(Response response) {
+            return response;
+        }
+
+        @Override
+        public Optional<String> accepts() {
+            return Optional.empty();
+        }
+    }
+
+    /** Implements exception handling equivalent dialogue decoders. */
+    static final class RemoteExceptionDecoder implements feign.codec.ErrorDecoder {
+        static final feign.codec.ErrorDecoder INSTANCE = new RemoteExceptionDecoder();
+
+        private static final ObjectMapper MAPPER = ObjectMappers.newClientObjectMapper();
+
+        private static final feign.codec.ErrorDecoder defaultDecoder = new feign.codec.ErrorDecoder.Default();
+
+        private boolean isError(feign.Response response) {
+            return 300 <= response.status() && response.status() <= 599;
+        }
+
+        private Exception parse(feign.Response response) {
+            String body;
+            try {
+                body = toString(response.body().asInputStream());
+            } catch (NullPointerException | IOException e) {
+                UnknownRemoteException exception = new UnknownRemoteException(response.status(), "<unparseable>");
```

Comment:
I _think_ if we wanted to exactly match the old behaviour, we'd pass in null for the second arg if the response.body() is null rather than the magic string `"<unparseable>"`?

Reply:
The okhttp UnknownRemoteExceptionResponseHandler uses `"<empty>"` and `"Failed to read response body"`. I can match those.